app/Http/Controllers/Api/ParticipationProject/ParticipationProjectController.php

```diff
@@ -552,34 +553,47 @@ public function checkDoubleAddress(Project $project, $contactId, $postalCodeNumb
     public function validatePostalCode(&$message, Project $project, Contact $contact)
     {
         $checkText = 'Postcode check: ';
-        $primaryAddress = $contact->primaryAddress;
-        if(!$primaryAddress){
-            array_push($message, $checkText . 'Deelnemer heeft geen primair adres.');
+        $addressForPostalCodeCheck = $contact->addressForPostalCodeCheck;
+        if($contact->type_id === ContactType::ORGANISATION) {
+            $typeAddress ='bezoek adres';
+        }else{
+            $typeAddress ='primair adres';
+        }
+        if(!$addressForPostalCodeCheck){
+            array_push($message, $checkText . 'Deelnemer heeft geen ' . $typeAddress. '.');
```

Comment:
Voor array_push, gebruik (overal) $message[] = (waarde) 
Dit is sneller in process time

Reply:
Dit heb je wel in AddressController gedaan 👍🏼